src/Sylius/Bundle/CoreBundle/Controller/ProductController.php

```diff
@@ -114,6 +115,38 @@ public function indexByTaxonIdAction(Request $request, $id)
         return $this->renderResults($taxon, $paginator, 'productIndex.html', $request->get('page', 1));
     }
 
+    /**
+     * List products categorized under given archetype (fetch by its code or id).
+     *
+     * @param Request $request
+     *
+     * @return Response
+     *
+     * @throws NotFoundHttpException
+     */
+    public function indexByArchetypeAction(Request $request)
+    {
+        $archetypeIdOrCode = $request->get('archetype');
```

Comment:
as it's mentionned in the request class, using `$request->get` is not recommended. https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/HttpFoundation/Request.php#L709
use $request->query->get or $request->request->get if get or post parameter.